File: ios/TestModule.m
#import "TestModule.h"

@implementation TestModule

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(NavigateMe: (void))

RCT_EXPORT_MODULE();

@end

File: ios/TestModule.swift
@objc(TestModule)
class TestModule: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }

    @objc(NavigateMe)
    func NavigateMe() -> Void {
        let vc = TestController(nibName: "TestController", bundle: nil)
        vc.present(vc, animated: true)
    }
}
